searchlib/src/vespa/searchlib/attribute/reference_mappings.h

```diff
@@ -86,7 +86,9 @@ class ReferenceMappings
         std::atomic_thread_fence(std::memory_order_acquire);
         return TargetLids(&_targetLids[0], committedDocIdLimit);
     }
-    uint32_t getTargetLid(uint32_t doc) const { return _targetLids[doc]; }
+    uint32_t getTargetLid(uint32_t doc) const {
+        return doc < _committedDocIdLimit ? _targetLids[doc] : 0u;
```

Comment:
Consider making a unit test for this. E.g in reference_attribute_test.cpp, as ReferenceAttribute uses ReferenceMappings. Would also be nice with a comment on why this is needed.